benchmark/result/result.py

```diff
@@ -0,0 +1,48 @@
+"""Basic result classes for benchmarking."""
+
+from typing import Any
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+
+@define(frozen=True)
+class Result:
```

Comment:
Given that the ability to store a `Result` object is central, can you elaborate on why you do not simply use the existing `SerialMixin` class? That should give you serializability out of the box, did you consider this?

Reply:
I dont think this will be necessary, our serialization engine already supports dataframe

See for isntance the [custom parameter](https://github.com/emdgroup/baybe/blob/main/baybe/parameters/custom.py#L34) which features a dataframe

[Here](https://github.com/emdgroup/baybe/blob/main/baybe/serialization/core.py#L92-L114) the code of our dataframe de/serializer and [where its registered](https://github.com/emdgroup/baybe/blob/main/baybe/serialization/core.py#L164-L165). We store them base encoded but deserialization can also happen via a custom format indicating a pandas constructor, [here](https://github.com/emdgroup/baybe/blob/main/tests/serialization/test_dataframe_serialization.py#L20-L59) examples how thats used in different variants

If you require de/serialization of a new type, custom functions can be registered for that, but no need to overwrite `to_` functions

Reply:
Cool, the hook was what I've looking for. Thanks, will add the inheritence and a hook for the uuid.